app/components/ucsf/RefinementLists/EligibilityRefinements.tsx

```diff
@@ -21,6 +21,7 @@ export const EligibilityRefinements = ({
   setSelectedEligibilities: (categories: SelectedEligibilities) => void;
   resourceSlug: string;
 }) => {
+  // Why is eligibilityGroupList a state variable and not just a prop?
   useEffect(() => {
     setEligibilityGroupList(resourceEligibilityGroups);
   }, [resourceSlug]);
```

Comment:
This line is causing the last remaining error, due to `resourceEligibilityGroups` not being present in the dependencies list. As written today, this hook only fires when `resourceSlug` is changed, or on the very first render of this component after it is mounted.

I'm not sure if adding `resourceEligibilityGroups` to the dependencies list is safe, in that I don't know if we were relying on only actually updating the `resourceEligibilitiesGroups` when the `resourceSlug` is also updated at the same time.

I tried reading the surrounding code, and I'm a little bit confused by what this `useEffect()` call is trying to accomplish in the first place. It might just be a bit of weird, vestigial code arising from multiple independent refactorings, and unfortunately, the git history is also difficult to follow because this code has come from multiple splicing/desplicing and file renames.

The main weirdness to me is that `setEligibilityGroupList()` is only ever called here in this `useEffect()` block. It's not clear to me why this even needs to be a state variable in the first place, since it's basically only initialized whenever `resourceSlug` changes. `resourceEligibilityGroups` itself is a prop of the this component, and there's some nontrivial code in `UcsfDiscoveryForm.tsx` that produces and operates on it.

@schroerbrian, since you've touched this code a lot lately, I'm wondering if you could provide a bit of insight here? Without completely grokking the calling code, I think the simplest fix would be to remove the `eligibilityGroupList` state variable from this component and instead just use the `resourceEligibilityGroups` prop in its place. However, I don't know if there's something more nuanced going on here, so any knowledge you have about this code would be great hear!

Reply:
Thanks for catching this, Richard! I really can't recall why I used `useEffect` here though I vaguely remember there being a reason for this. I switched to using the `resourceEligiblityGroups` prop locally, and it seems to be working fine. I have created a [PR](https://github.com/ShelterTechSF/askdarcel-web/pull/1213/files) into this branch that makes the switch.